src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -149,66 +149,13 @@ public static bool InStandbyMode
             }
         }
 
-        public IReadOnlyDictionary<IHttpRoute, FunctionDescriptor> HttpFunctions
-        {
-            get
-            {
-                return _httpFunctions as IReadOnlyDictionary<IHttpRoute, FunctionDescriptor>;
-            }
-        }
-
-        public async Task<HttpResponseMessage> HandleRequestAsync(FunctionDescriptor function, HttpRequestMessage request, CancellationToken cancellationToken)
-        {
-            // ensure that the host is ready to process requests
-            await DelayUntilHostReady(_hostTimeoutSeconds, _hostRunningPollIntervalMilliseconds);
-
-            // All authentication is assumed to have been done on the request
-            // BEFORE this method is called
-            var arguments = GetFunctionArguments(function, request);
-
-            // Suspend the current synchronization context so we don't pass the ASP.NET
-            // context down to the function.
-            using (var syncContextSuspensionScope = new SuspendedSynchronizationContextScope())
-            {
-                // Add the request to the logging scope. This allows the App Insights logger to
-                // record details about the request.
-                ILoggerFactory loggerFactory = _config.HostConfig.GetService<ILoggerFactory>();
-                ILogger logger = loggerFactory.CreateLogger(LogCategories.Function);
-                var scopeState = new Dictionary<string, object>()
-                {
-                    [ScriptConstants.LoggerHttpRequest] = request
-                };
-                using (logger.BeginScope(scopeState))
-                {
-                    await Instance.CallAsync(function.Name, arguments, cancellationToken);
-                }
-            }
-
-            // Get the response
-            HttpResponseMessage response = null;
-            if (!request.Properties.TryGetValue<HttpResponseMessage>(ScriptConstants.AzureFunctionsHttpResponseKey, out response))
-            {
-                // the function was successful but did not write an explicit response
-                response = new HttpResponseMessage(HttpStatusCode.OK);
-            }
-
-            return response;
-        }
-
-        public void Initialize()
+        public void Initialize(CancellationToken cancellationToken)
         {
             lock (_syncLock)
             {
                 if (!_hostStarted)
                 {
-                    if (!_webHostSettings.IsSelfHost)
-                    {
-                        HostingEnvironment.QueueBackgroundWorkItem((ct) => RunAndBlock(ct));
```

Comment:
We used to do things differently based on whether this was self host. Why aren't we queueing this up with the HostingEnvironment anymore?

Reply:
This API no longer exists. This is now handled by the `WebJobsScriptHostService`, which is an `IHostedService` registered with ASP.NET